src/train.py

```diff
@@ -0,0 +1,356 @@
+    :param tokenizer: path to pretrained tokenizer directory. If None, vtr model will be trained (default: None)
+    :param max_seq_len: used to create datasets. In the case of vtr, a limit on the number of slices, in the case of a classic transformer, on the number of tokens. (default: 512)
+    :param max_slices_count_per_word: used to limit the number of slices for one word. You can learn more about the process of processing a sequence of words by looking at the image `resources/images/padding_in_sl.jpg` (default: 9)
+    :param font_size: font size with which the image is generated (default: 15)
+    :param window_size: window width (default: 30)
+    :param stride: window step size (default: 5)
+    :param epochs: number of training epochs (default: 10)
+    :param random_state: (default: 21)
+    :param log_every: how often to recalculate the value of metrics on the validation set. These values are logged to wandb (default: 1000)
+    :param emb_size: embedding size received by visual representations (default: 768)
+    :param num_workers: number of threads in DataLoader (default: 1)
+    :param save_to: the path where the weights of the model are saved after training, if a value is specified, if not, then the weights are not saved (default: None)
+    :param font: font used to generate character images (default: resources/fonts/NotoSans.ttf)
+    :param out_channels: number of output channels in the last convolutional layer (default: 256)
+    :param nhead: number of transformer layer heads (default: 12)
+    :param beta1: Adam optimizer parameter (default: 0.9)
+    :param beta2: Adam optimizer parameter (default: 0.999)
+    :param sl: if true, then the task of labeling the sequence is being solved (default: False)
+    :param dropout: dropout probability (default: 0.0)
+    :param device: 'cuda' or 'cpu'. If None it will be set automatically (default: None)
+    :param warmup: number of steps to warm up Linear Scheduler (default: 1000)
+    :param batch_size: batch size (default: 32)
+    :param kernel_size: convolution kernel size (default 3)
+    :param num_layers: number of transformer encoder layers in the visual representation model (default: 1)
+    :param lr: learning rate (default: 5e-5)
+    :return:
+    """
+    wandb.init(project="visual-text", entity="borisshapa")
+    set_deterministic_mode(random_state)
+
+    actual_device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+    labeled_texts = load_json(train_data)
+
+    val_labeled_texts = None
+    if val_data is not None:
+        val_labeled_texts = load_json(val_data)
+
+    test_labeled_texts = None
+    if test_data is not None:
+        test_labeled_texts = load_json(test_data)
+
+    model: nn.Module
+    criterion: Callable[[torch.Tensor, torch.Tensor], torch.Tensor]
+
+    dataset: SizedCollatedDataset
+    val_dataset: Optional[SizedCollatedDataset] = None
+    test_dataset: Optional[SizedCollatedDataset] = None
```

Comment:
I really don't like this, it is very complicated and not pythonic at all.

I suggest to add functions to init modules:
```python
def init_sl_pipeline(some args) -> Tuple[nn.Module, LOSS]
    ...

def init_classic_pipeline(some args) -> Tuple[nn.Module, LOSS]
    ...
```

And then in the main code:
```python
if sl:
    model, criterion = init_sl_pipeline()
else:
   model, criterion = init_classic_pipeline()
```

In the same way, you can define logic to init datasets. And with such logic, you can omit this pasta with type definitions.